src/Controls/src/Core/Label/Label.iOS.cs

```diff
@@ -1,11 +1,36 @@
 ﻿#nullable disable
 using System;
 using Microsoft.Maui.Controls.Platform;
+using UIKit;
 
 namespace Microsoft.Maui.Controls
 {
 	public partial class Label
 	{
+		MauiLabel _mauiLabel;
+
+		private protected override void OnHandlerChangedCore()
+		{
+			base.OnHandlerChangedCore();
+
+			if (Handler is not null)
+			{
+				if (Handler is LabelHandler labelHandler && labelHandler.PlatformView is MauiLabel mauiLabel)
+				{
+					_mauiLabel = mauiLabel;
+					_mauiLabel.LayoutSubviewsChanged += OnLayoutSubviewsChanged;
+				}
+			}
+			else
+			{
+				if (_mauiLabel is not null)
+				{
+					_mauiLabel.LayoutSubviewsChanged -= OnLayoutSubviewsChanged;
```

Comment:
Reviewing failing tests. The test that fails is because this is never invoked, and produces a leak detected by one of the tests (`PagesDoNotLeak`).